lib/authentication/AuthenticationController.ts

```diff
@@ -0,0 +1,83 @@
+import * as bcrypt from "bcrypt";
+import {Request , Response, NextFunction} from "express";
+import * as jwt from "jsonwebtoken";
+import CreateUserDto from "dataTransferObjects/CreateUserDto";
+import CustomerWithThatEmailAlreadyExistsException from "../exceptions/CustomerWithThatEmailAlreadyExists";
+import WrongLoginCredentialsException from "../exceptions/WrongLoginCredentialsException";
+import LogInDto from "../dataTransferObjects/LogInDto"
+import TokenData from "./TokenData";
+import DataStoredInToken from "./DataStoredInToken";
+import { UserModel, User, IUser } from "../models/userModel";
+
+class AuthenticationController {
+    
+    private model : UserModel;
+    private static controller : AuthenticationController;
+
+    private constructor(){
+        this.model = new User().model;
+    }
+
+    public static getInstance = () => {
+        if(!AuthenticationController.controller){
+            AuthenticationController.controller = new AuthenticationController();
+        }
+        return AuthenticationController.controller;
+    }
+
+    public registration = async (request : Request, response : Response, next: NextFunction) => {
+        const userData : CreateUserDto =  request.body; 
+        console.log(userData);
+        if(await this.model.findOne({ email: userData.email})){
+            next(new CustomerWithThatEmailAlreadyExistsException(userData.email))
+        }else{
```

Comment:
du brauchst hier das else nicht. Wenn die exception geschmissen wird. Sollte der part dadrunter sowieso nicht mehr ausgeführt werden. Versuche in so welchen Situationen auf else zu verzichten.

Reply:
stimmt, guard clauses verwenden^